core/src/main/scala/zio/config/ReadModule.scala

```diff
@@ -232,24 +239,61 @@ private[config] trait ReadModule extends ConfigDescriptorModule {
       path: List[Step[K]],
       keys: List[K],
       config: ConfigDescriptor[B],
-      descriptions: List[String]
+      descriptions: List[String],
+      programSummary: List[ConfigDescriptor[_]]
```

Comment:
An early exit should be only for recursive values (`programSummary.contains(config)`)
  Such that if configuration isn't recursive, we report all errors at deeper levels.
  
  Example:
  Without a program summary, for a non-recursive structure, we get the following error structure:
 ```scala
     MissingValue
      path: a.a
      Details: optional value
   ```
  
  With program summary, for a non-recursive structure, we get the following error structure
  
  ```scala
    MissingValue
      path: a.a.b2
      Details: optional value, value of type string
    MissingValue
      path: a.a.a2
      Details: optional value, value of type string
  ```